src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -158,6 +158,15 @@ internal static async Task<NameDeclarationInfo> GetDeclarationInfoAsync(Document
                 SyntaxToken token, SemanticModel semanticModel,
                 CancellationToken cancellationToken, out NameDeclarationInfo result)
             {
+                // Incomplete statements like "SomeSymbol " or "SomeEvent " are syntactically equal.
+                // However you cannot declare a variable or local function with event being a return type,
+                // so we need to filter this case here
```

Comment:
I believe this is not the approach we were expecting. We only wanted to suppress the name suggestion if the qualified expression on the left started with `this.` (or `Me.`). The type of symbol (event, property, etc.) is not relevant for this check; it is a syntax-only check.

Reply:
@DoctorKrolic this is partially our fault for not verifying this bug report. We had an internal email discussion where we arrived at the conclusion that `this.` was relevant, but that information was omitted from the public report and I can see now how that was misleading. I apologize for missing that.